frontend/src/components/TreeViewer/TreeViewer.jsx

```diff
@@ -182,28 +189,70 @@ const setOpacity = (data, link, currentOpacity) => {
  * Note: The name of the node is a key within the particular tree
  *
  */
-function createNameToNodeMapping(currNode, mapping = {}) {
+function createNameToNodeMapping(currNode, mapping = {}, parent = "") {
+  currNode.parent = parent;
   mapping[currNode.name] = currNode;
   if (currNode.children) {
     currNode.children.forEach((child) =>
-      createNameToNodeMapping(child, mapping)
+      createNameToNodeMapping(child, mapping, currNode.name)
     );
   }
   return mapping;
 }
 
-/* Traverse from currNode to the ancestor node called ancestorNodeName
- *  and collect all the node names along the path.
+/**
+ * Traverse from currNode to the ancestor node called ancestorNodeName
+ * and collect all the node names along the path.
+ * @param {Node} currNode : The node to start from
+ * @param {dict} nameToNodeMapping : A mapping of node names to nodes
+ * @param {*} ancestorNodeName : The node to stop the path at; if not given,
+ * finds the complete path to the root
+ * @param {*} history : The nodes in the path found so far. Used only in recursive
+ * call
+ * @returns {List{string}} representing list of names from currNode to ancestorNode
  */
-function pathToAncestor(currNode, ancestorNodeName, history = []) {
-  if (currNode && currNode.data.name !== ancestorNodeName) {
-    history.push(currNode.data.name);
-    pathToAncestor(currNode.parent, ancestorNodeName, history);
+function pathToAncestor(
```

Comment:
Probably best to convert this to iterative since we can keep going to the parent while it is not **ancestorNodeName**

Reply:
Question: Is there a disadvantage to recursive? Just wondering, since this function has been initially recursive and I don't think that conditions have changed?

Reply:
Yea, I made it initially recursive but I think it'll be a lot more readable if its iterative. With recursive, the call stack will also grow with every recursive call, but that's not a big deal at this scale.

Reply:
Ok, fixed